### PR TITLE
Make SimulationServer class reusable

### PR DESCRIPTION
**Pull request by mfehr**
This PR doesn't change anything in the simulator logic, it's simply a move and slight architecture change. Changes:
 * Cleanup of ros params in simulation server.
 * Moves the simulation server from simulation_eval.cc to a separate header and source file to make it reusable.
 * To use it one now simply implements a derived class and the prepareWorld function, which defines how the world looks like.

### CODE REVIEW

File: voxblox_ros/src/simulation_server.cc [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+                       getEsdfIntegratorConfigFromRosParam(nh_private),
+                       getTsdfMapConfigFromRosParam(nh_private),
+                       getTsdfIntegratorConfigFromRosParam(nh_private)) {}
+
+bool SimulationServer::generatePlausibleViewpoint(FloatingPoint min_distance,
+                                                  Point* ray_origin,
+                                                  Point* ray_direction) const {
+  // Generate a viewpoint at least min_distance from any objects (if you want
+  // just outside an object, just call this with min_distance = 0).
+  constexpr int max_tries = 50;
```
Comment (ZacharyTaylor) [inline voxblox_ros/src/simulation_server.cc]: could we expose this or at least move the 50 out of being hidden in the function?

File: voxblox_ros/src/simulation_server.cc [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+bool SimulationServer::evaluateVoxel(const TsdfVoxel& voxel_test,
+                                     const TsdfVoxel& voxel_gt,
+                                     FloatingPoint* error) const {
+  if (voxel_test.weight < 1e-6 || voxel_gt.distance < 0.0) {
+    return false;
+  }
+
+  *error = voxel_gt.distance - voxel_test.distance;
+
+  /* if (voxel_gt.distance < -truncation_distance_) {
```
Comment (ZacharyTaylor) [inline voxblox_ros/src/simulation_server.cc]: why is this commented out?

File: voxblox_ros/src/simulation_server.cc [hunk 3]
```diff
@@ -0,0 +1,417 @@
...
+  ROS_INFO_STREAM("TSDF RMSE: " << tsdf_rmse << " ESDF RMSE: " << esdf_rmse);
+
+  ROS_INFO_STREAM("Mesh Timings: " << std::endl << timing::Timing::Print());
+}
+
+void SimulationServer::visualize() {
+  if (!visualize_) {
+    return;
+  }
+  FloatingPoint slice_level = 2.0;
```
Comment (ZacharyTaylor) [inline voxblox_ros/src/simulation_server.cc]: Could this be a parameter, may not always want 2 meters

File: voxblox_ros/src/simulation_server.cc (hunk 2 above)
Comment (helenol) [reply voxblox_ros/src/simulation_server.cc]: Ahh I think this should be deleted -- this has to do with how behind-surface measurements are penalized in the evaluation, but I eventually settled on only evaluating positive distances (I think this is mentioned in the paper) and the if statement above confirms this. So yeah should just be deleted, here and below.

**Pull request merged**